vividus-to-azure-devops-exporter/src/main/java/org/vividus/azure/devops/client/AzureDevOpsClient.java

```diff
@@ -19,58 +19,120 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
+import java.util.Set;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 
+import org.apache.http.HttpEntity;
 import org.apache.http.client.methods.HttpRequestBase;
 import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.vividus.azure.devops.client.model.AddOperation;
+import org.vividus.azure.devops.client.model.Entity;
+import org.vividus.azure.devops.client.model.PointsQuery;
+import org.vividus.azure.devops.client.model.TestPoint;
+import org.vividus.azure.devops.client.model.TestPointContainer;
+import org.vividus.azure.devops.client.model.TestResult;
+import org.vividus.azure.devops.client.model.TestRun;
+import org.vividus.azure.devops.client.model.WorkItem;
 import org.vividus.azure.devops.configuration.AzureDevOpsExporterOptions;
 import org.vividus.http.HttpMethod;
 import org.vividus.http.HttpRequestBuilder;
 import org.vividus.http.client.IHttpClient;
+import org.vividus.util.json.JsonUtils;
 
 public class AzureDevOpsClient
 {
     private static final Logger LOGGER = LoggerFactory.getLogger(AzureDevOpsClient.class);
 
     private static final String API_VERSION = "?api-version=7.1-preview.2";
 
+    private static final String WORK_ITEMS_PART = "wit/workitems";
+    private static final String FORWARD_SLASH = "/";
+
     private final IHttpClient httpClient;
+    private final JsonUtils jsonUtils;
     private final String endpoint;
-    private final ObjectMapper mapper;
     private final String basePath;
 
     public AzureDevOpsClient(IHttpClient httpClient, AzureDevOpsExporterOptions options)
     {
         this.httpClient = httpClient;
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        mapper.findAndRegisterModules();
+        this.jsonUtils = new JsonUtils(mapper);
         this.endpoint = "https://dev.azure.com/";
-        this.mapper = new ObjectMapper();
-        this.basePath = String.format("/%s/%s/_apis/wit/workitems", options.getOrganization(), options.getProject());
+        this.basePath = String.format("/%s/%s/_apis/", options.getOrganization(), options.getProject());
+    }
+
+    public WorkItem createTestCase(List<AddOperation> operations) throws IOException
+    {
+        return executeOnWorkItem(HttpMethod.POST, "/$test case", operations);
+    }
+
+    public WorkItem getWorkItem(Integer workItemId) throws IOException
+    {
+        return asWorkItem(execute(HttpMethod.GET, basePath + WORK_ITEMS_PART + FORWARD_SLASH + workItemId, null));
```

Comment:
I propose to store this as a pattern, so you don't have to "concatenate" to see the full endpoint

Reply:
decided to move concatenation of basePath + WORK_ITEMS_PART + FORWARD_SLASH into separate object variable keeping concatenation simpler and more concise

Reply:
It's just not readable enough from my POV